File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int ans=nums[0];
        int n = nums.size();
        int last = nums[0];
        for(int i=1;i<n;i++)
         {
            int curr = max(nums[i],nums[i]+last);
            ans = max(ans,curr);
            last = curr;
         }
        return ans;
    }
};

File: Data Structures & Algorithms/pow-x-n/submission-1.java
class Solution {
    public double myPow(double x, int n) {
        if(n<0)
         return 1/myPow(x,-1*n);
        if(n==0)
         return 1;
        double ans = myPow(x,n/2);
        return n%2==1?ans*ans*x:ans*ans;
    }
}
